osu.Game/Stores/RealmArchiveModelManager.cs

```diff
@@ -45,11 +45,16 @@ private void performFileOperation(TModel item, Action<TModel> operation)
             // This method should be removed as soon as all the surrounding pieces support non-detached operations.
             if (!item.IsManaged)
             {
-                var managed = Realm.Realm.Find<TModel>(item.ID);
-                managed.Realm.Write(() => operation(managed));
-
-                item.Files.Clear();
-                item.Files.AddRange(managed.Files.Detach());
+                // Importantly, begin the realm write *before* re-fetching, else the update realm may not be in a consistent state
+                // (ie. if an async import finished very recently).
+                Realm.Realm.Write(realm =>
+                {
+                    var managed = Realm.Realm.Find<TModel>(item.ID);
```

Comment:
Also, bit of a nitpick but this can now use the lambda param rather than capturing outwardly, I think?
```suggestion
                    var managed = realm.Find<TModel>(item.ID);
```